bundle/edu.gemini.spModel.core/src/main/scala/edu/gemini/spModel/core/MagnitudeSystem.scala

```diff
@@ -1,14 +1,25 @@
 package edu.gemini.spModel.core
 
+import scala.collection.JavaConversions._
 import scalaz.Equal
 
-sealed abstract class MagnitudeSystem(val name: String) extends Product with Serializable
+sealed abstract class MagnitudeSystem(val name: String) extends Product with Comparable[MagnitudeSystem] with Serializable {
+
+  // ===== LEGACY JAVA SUPPORT =====
+  import MagnitudeSystem._
+
+  def compareTo(other: MagnitudeSystem) = all.indexOf(other) - all.indexOf(this)
```

Comment:
Again, this mimics the Java enum behavior by replacing the `ordinal()` method which is used for `compareTo` on enums, which by default is based  on the order the values are defined. There isn't really a natural order. The only place this is relevant is in the `compareTo()` method of the legacy `Magnitude` class. My main objective was to not change the sort order because I don't really know where it is relevant, but I am open for suggestions how to do this in a better way. Maybe I should move that code directly into the `compareTo()` method though, so that it is clear this is not really meant to be used for anything other than that.